src/main/java/edu/kit/ipd/crowdcontrol/objectservice/quality/QualityIdentificator.java

```diff
@@ -165,43 +168,37 @@ private void checkExpStatus(ExperimentRecord experiment) {
-            List<RatingRecord> records = answerOperations.getRatingsOfAnswer(answer);
+    private void rateQualityOfRatings(AnswerRecord answerRecord, Map<AlgorithmRatingQualityParamRecord, String> params) {
+           List<RatingRecord> records = answerRatingOperations.getRelatedRatings(answerRecord.getIdAnswer());
             Map<RatingRecord, Integer> map = ratingIdentifier.identifyRatingQuality(records, params, MAXIMUM_QUALITY, MINIMUM_QUALITY);
-            answerOperations.setQualityToRatings(map);
-        }
-
+            answerRatingOperations.setQualityToRatings(map);
     }
 
 
     /**
      * Rates and sets quality of all answers of specified experiment.
-     * Only uses ratings of a specified quality (@ratingQualityThreshold)
+     * Only uses ratings of a specified quality
      * Furthermore checks if a specified amount of ratings is present for that answer
      * and it thus the answer's quality is unlikely to change. In that case the corresponding
      * quality-assured-bit is set in the database.
      *
-     * @param experimentRecord the experiment whose answers are going to be rated
+     * @param answerRecord  which is going to be rated
+     * @param params Mapping of parameter-records to the user specified parameters represented as a string
      */
-    private void rateQualityOfAnswers(ExperimentRecord experimentRecord, Map<AlgorithmAnswerQualityParamRecord, String> params) {
-
-        List<AnswerRecord> answers = answerOperations.getAnswersOfExperiment(experimentRecord.getIdExperiment());
+    private void rateQualityOfAnswers(AnswerRecord answerRecord, Map<AlgorithmAnswerQualityParamRecord, String> params) {
         Map<String, Integer> result;
-        for (AnswerRecord answer : answers) {
-            result = answerIdentifier.identifyAnswerQuality(answerOperations, answer, params, MAXIMUM_QUALITY, MINIMUM_QUALITY);
-            answerOperations.setQualityToAnswer(answer, result.get(AnswerQualityStrategy.QUALITY));
+        result = answerIdentifier.identifyAnswerQuality(answerRatingOperations, answerRecord, params, MAXIMUM_QUALITY, MINIMUM_QUALITY);
+        answerRatingOperations.setQualityToAnswer(answerRecord, result.get(AnswerQualityStrategy.QUALITY));
 
-            // Checks if quality_assured bit can be set.
-            if ((((double) result.get(AnswerQualityStrategy.NUM_OF_RATINGS) / (double) experimentRecord.getRatingsPerAnswer()) >= 0.8)) {
-                answerOperations.setAnswerQualityAssured(answer);
-            }
+        // Checks if quality_assured bit can be set.
+        if ((((double) result.get(AnswerQualityStrategy.NUM_OF_RATINGS) / (double) experimentOperations.getExperiment(answerRecord.getExperiment())
+                .orElseThrow(() -> new IllegalArgumentException("Error, can't find experiment of given experiment-ID: "+answerRecord.getExperiment())).getRatingsPerAnswer()) >= 0.8)) {
+            answerRatingOperations.setAnswerQualityAssured(answerRecord);
```

Comment:
Wasn't the quality_assured_bit to be meant to indicate the presence of a certain amount of ratings for an answer?